crates/prover-engine/src/lib.rs

```diff
@@ -22,6 +22,7 @@ pub struct ProverEngine {
     cancellation_token: Option<CancellationToken>,
     metric_socket_addr: Option<SocketAddr>,
     rpc_socket_addr: Option<SocketAddr>,
+    shutdown_runtime_timeout: Option<Duration>,
```

Comment:
Nit: the following feels a bit more natural to me:

```suggestion
    runtime_shutdown_timeout: Option<Duration>,
```